r/tests/testthat/helper-data.R

```diff
@@ -134,3 +134,17 @@ example_with_logical_factors <- tibble::tibble(
     "hey buddy"
   )
 )
+
+# the values in each column of this tibble are in ascending order in the C locale.
+# there are some ties, but sorting by any two columns will give a deterministic order.
+example_data_for_sorting <- tibble::tibble(
+  int = c(-.Machine$integer.max, -101L, -100L, 0L, 0L, 1L, 100L, 1000L, .Machine$integer.max, NA_integer_),
+  dbl = c(-Inf, -.Machine$double.xmax, -.Machine$double.xmin, 0, .Machine$double.xmin, pi, .Machine$double.xmax, Inf, NaN, NA_real_),
+  # R string collation varies by locale, while libarrow always uses the C locale for string collation
+  # (in other words: string values in libarrow are ordered lexicographically as bytestrings)
+  # to make R sort functions use the C locale, run Sys.setlocale("LC_COLLATE", "C")
```

Comment:
Doesn't testthat set that?

Reply:
Does it? I don't know

Reply:
I'm pretty sure it does now; if it's a concern, we can set it in `helper-arrow.R` to be explicit. But we should have deterministic sort order in our test suite, for sure.

Reply:
Confirmed it does, and this is mentioned at https://testthat.r-lib.org/reference/local_test_context.html#details. I improved this comment in 7b64d1ff18b4f587313be212fc0816e55a54a754.